frontend/lib/messages/model/message.model.dart

```diff
@@ -1,15 +1,79 @@
 class MessageModel {
+  final List<Conversation> conversations;
+
+  MessageModel({this.conversations = const []});
+
+  factory MessageModel.fromJson(Map<String, dynamic> json) {
+    return MessageModel(
+      conversations: (json['conversations'] as List?)
+              ?.map((conv) => Conversation.fromJson(conv))
+              .toList() ??
+          [],
+    );
+  }
+}
+
+class Conversation {
+  final String id;
+  final String userId;
+  final String userName;
+  final String userImageUrl;
   final List<Message> messages;
+  final DateTime lastMessageTime;
+  final bool isUnread;
 
-  MessageModel({this.messages = const []});
+  Conversation({
+    required this.id,
+    required this.userId,
+    required this.userName,
+    required this.userImageUrl,
+    required this.messages,
+    required this.lastMessageTime,
+    this.isUnread = false,
+  });
+
+  String get lastMessagePreview {
+    if (messages.isEmpty) return '';
+    return messages.last.content.length > 30
+        ? '${messages.last.content.substring(0, 30)}...'
+        : messages.last.content;
+  }
+
+  factory Conversation.fromJson(Map<String, dynamic> json) {
+    return Conversation(
+      id: json['id'],
+      userId: json['user_id'],
+      userName: json['user_name'],
+      userImageUrl: json['user_image_url'],
+      messages: (json['messages'] as List?)
+              ?.map((msg) => Message.fromJson(msg as Map<String, dynamic>))
+              .toList() ??
+          [],
+      lastMessageTime: DateTime.parse(json['last_message_time']),
+      isUnread: json['is_unread'] == true,
+    );
+  }
 }
 
 class Message {
   final String id;
   final String content;
+  final DateTime timestamp;
+  final bool isFromCurrentUser;
 
   Message({
     required this.id,
     required this.content,
+    required this.timestamp,
+    required this.isFromCurrentUser,
   });
-}
+
+  factory Message.fromJson(Map<String, dynamic> json) {
+    return Message(
+      id: json['id'],
+      content: json['content'],
+      timestamp: DateTime.parse(json['timestamp']),
+      isFromCurrentUser: json['is_from_current_user'],
```

Comment:
This field will likely not come from the backend (which only provides the sender and receiver) but it's easy enough to add simple boolean logic on the client side later that derives this.